site/gdocs/ArticleBlock.tsx

```diff
@@ -0,0 +1,95 @@
+import React from "react"
+
+import ChartStory from "./ChartStory"
+import Scroller from "./Scroller"
+import Chart from "./Chart"
+import PullQuote from "./PullQuote"
+import FixedGraphic from "./FixedGraphic"
+import Recirc from "./Recirc"
+import List from "./List"
+import Image from "./Image"
+import { OwidArticleBlock } from "../../clientUtils/owidTypes.js"
+
+export default function ArticleBlock({ d }: { d: OwidArticleBlock }) {
+    const handleArchie = (d: OwidArticleBlock, key: string) => {
+        const _type = d.type.toLowerCase()
+        let content: any = JSON.stringify(d)
+        if (_type === "chart") {
+            content = <Chart d={d} key={key} />
+        } else if (_type === "aside") {
+            content = (
+                <figure
+                    key={key}
+                    className={`aside-text ${
+                        d.value.position ? d.value.position : ""
+                    }`}
+                >
+                    {d.value.caption ? (
+                        <figcaption>{d.value.caption}</figcaption>
+                    ) : null}
+                </figure>
+            )
+        } else if (_type === "scroller") {
+            content = <Scroller key={key} d={d} />
+        } else if (_type === "chart-grid") {
+            let columns = 1
+            try {
+                columns =
+                    +d.value.find(
+                        (_d: OwidArticleBlock) => _d.type === "columns"
+                    ).value || 1
+            } catch (e) {}
+
+            return (
```

Comment:
Any reason for charts early-returning when everything else re-assigns `content`?

This block seems a little terse overall, I wouldn't mind some friendlier function names to help explain it.